ElementX/Sources/FlowCoordinators/PinnedEventsTimelineFlowCoordinator.swift

```diff
@@ -0,0 +1,161 @@
+                case .presentLocationViewer(let geoURI, let description):
+                    presentMapNavigator(geoURI: geoURI, description: description)
+                case .displayMessageForwarding(let forwardingItem):
+                    presentMessageForwarding(with: forwardingItem)
+                case .displayRoomScreenWithFocussedPin(let eventID):
+                    actionsSubject.send(.displayRoomScreenWithFocussedPin(eventID: eventID))
+                }
+            }
+            .store(in: &cancellables)
+        
+        navigationStackCoordinator.setRootCoordinator(coordinator)
+    }
+    
+    private func presentMapNavigator(geoURI: GeoURI, description: String?) {
+        let stackCoordinator = NavigationStackCoordinator()
+        
+        let params = StaticLocationScreenCoordinatorParameters(interactionMode: .viewOnly(geoURI: geoURI, description: description), appMediator: appMediator)
+        let coordinator = StaticLocationScreenCoordinator(parameters: params)
+        
+        coordinator.actions.sink { [weak self] action in
+            guard let self else { return }
+            switch action {
+            case .selectedLocation:
+                // We don't handle the sending/picker case in this flow
+                break
+            case .close:
+                self.navigationStackCoordinator.setSheetCoordinator(nil)
+            }
+        }
+        .store(in: &cancellables)
+        
+        stackCoordinator.setRootCoordinator(coordinator)
+        
+        navigationStackCoordinator.setSheetCoordinator(stackCoordinator)
+    }
+    
+    private func presentMessageForwarding(with forwardingItem: MessageForwardingItem) {
+        guard let roomSummaryProvider = userSession.clientProxy.alternateRoomSummaryProvider else {
+            fatalError()
+        }
+        
+        let stackCoordinator = NavigationStackCoordinator()
+        
+        let parameters = MessageForwardingScreenCoordinatorParameters(forwardingItem: forwardingItem,
+                                                                      clientProxy: userSession.clientProxy,
+                                                                      roomSummaryProvider: roomSummaryProvider,
+                                                                      mediaProvider: userSession.mediaProvider,
+                                                                      userIndicatorController: userIndicatorController)
+        let coordinator = MessageForwardingScreenCoordinator(parameters: parameters)
+        
+        coordinator.actions.sink { [weak self] action in
+            guard let self else { return }
+            
+            switch action {
+            case .dismiss:
+                navigationStackCoordinator.setSheetCoordinator(nil)
+            case .sent(let roomID):
+                navigationStackCoordinator.setSheetCoordinator(nil)
+                actionsSubject.send(.startChildRoomFlow(roomID: roomID))
```

Comment:
This action should probably describe what happened and let the parent flow coordinator decide the actual action. E.g.
```suggestion
                actionsSubject.send(.forwardedMessageToRoom(roomID: roomID))
```